pkg/dao/public_repositories.go

```diff
@@ -0,0 +1,49 @@
+package dao
+
+import (
+	"github.com/content-services/content-sources-backend/pkg/models"
+	"gorm.io/gorm"
+)
+
+type PublicRepository struct {
+	UUID string
+	URL  string
+}
+
+func GetPublicRepositoryDao(db *gorm.DB) PublicRepositoryDao {
+	return publicRepositoryDaoImpl{
+		db: db,
+	}
+}
+
+type publicRepositoryDaoImpl struct {
+	db *gorm.DB
+}
+
+func (p publicRepositoryDaoImpl) FetchForUrl(url string) (error, PublicRepository) {
+	repo := models.Repository{}
+	result := p.db.Where("public = true and URL = ?", url).First(&repo)
+	if result.Error != nil {
+		return result.Error, PublicRepository{}
+	}
+	return nil, PublicRepository{
+		UUID: repo.UUID,
+		URL:  repo.URL,
+	}
+}
```

Comment:
Ryan's issue with the introspect command was with regard to this method only working with Public repos, while introspect-all works with public and user-defined.  We should probably make this consistent.    For now lets just change this to not care about public = true.  

I'm gonna file an issue to rename these dao structs to be more clear

Reply:
issue:  https://issues.redhat.com/browse/HMSCONTENT-145

Reply:
thanks! updated!